tools/azure-sdk-tools/devtools_testutils/resource_testcase.py

```diff
@@ -65,22 +69,29 @@ def create_resource(self, name, **kwargs):
             parameters = {"location": self.location}
             if self.delete_after_tag_timedelta:
                 expiry = datetime.datetime.utcnow() + self.delete_after_tag_timedelta
-                parameters["tags"] = {
-                    "DeleteAfter": expiry.replace(microsecond=0).isoformat()
-                }
+                parameters["tags"] = {"DeleteAfter": expiry.replace(microsecond=0).isoformat()}
+            try:
+                if "tags" not in parameters.keys():
+                    parameters["tags"] = {}
+                parameters["tags"]["BuildId"] = os.environ["BUILD_BUILDID"]
+                parameters["tags"]["BuildJob"] = os.environ["AGENT_JOBNAME"]
+                parameters["tags"]["BuildNumber"] = os.environ["BUILD_BUILDNUMBER"]
+                parameters["tags"]["BuildReason"] = os.environ["BUILD_REASON"]
+            except KeyError:
+                pass
             try:
-                self.resource = self.client.resource_groups.create_or_update(
-                    name, parameters
+                logging.info(
+                    "Attempting to create a Resource Group with name {} and parameters {}".format(name, parameters)
                 )
+                self.resource = self.client.resource_groups.create_or_update(name, parameters)
```

Comment:
Would it be possible to add a common prefix, something like "rgp-" to all the names? So we know that the resource groups are coming from here.